index.html

```diff
@@ -0,0 +1,317 @@
+
+Specs should attach their prefix to:
+- the names of any types, interfaces, unions, enums, or inputs defined in the spec
+- the names of any fields the spec introduces on *foreign types* defined in a different spec
+- the names of any directives introduced in the spec
+- specs may introduce one **root directive**, which carries *only* the prefix and no underscore.
+
+<view-of src=prefixing.graphql
+  title='Prefixing examples'
+></view-of>
+
+## Versioning
+
+Specs are versioned with a **subset** of a [Semantic Version Number](https://semver.org/spec/v2.0.0.html) containing only the `major` and `minor` parts. Thus, specifications **should** provide a version of the form `‹major›.‹minor›`, where both integers >= 0.
+
+```
+#::[Examples of valid versions]
+2.2
+1.0
+1.1
+0.1
+```
+
+As specified by semver, spec authors **should** increment the:
+> - MAJOR version when you make incompatible API changes,
+> - MINOR version when you add functionality in a backwards compatible manner
+
+Patch and pre-release qualifiers are judged to be not particularly meaningful in the context of specifications, which are (by definition) interfaces rather than implementations. The `‹patch›` component of version denotes a bug fix which is backwards compatible—that is, a change to the implementation which does not affect the interface. `‹patch›`-level changes in the version of a spec denote wording clarifications which do not require implementation changes. As such, it is not important to track them for the purposes of version resolution.
+
+As with [semver](https://semver.org/spec/v2.0.0.html), the `0.x` version series is special: there is no expectation of compatibility between versions `0.x` and `0.y`. For example, a processor must not activate implementation `0.4` to satisfy a requested version of `0.2`.
+
+### Ordering
+Versions can be ordered with the following psuedocode:
+
+```typescript
+compareVersions(a: Version, b: Version) -> LESS_THAN | EQUAL | GREATER_THAN {
+  if a.major < b.major {
+    return LESS_THAN
+  }
+  if a.major == b.major {
+    if a.minor == b.minor { return EQUAL }
+    if a.minor > b.minor { return GREATER_THAN }
+    return LESS_THAN
+  }
+  if a.major > b.major {
+    return GREATER_THAN
+  }
+}
+```
+
+Note that this ordering does *not* guarantee that if A > B then A can satisfy a version requirement of B. For example, `2.0 > 1.9`, but `2.0` cannot [satisfy](#satisfying-version-requirements) the requirements of a document which references version `1.9`, since the major version update introduced possibly-breaking changes.
+
+### Satisfaction
+
+Given a version `requested` by a document and an `available` version of an implementation, the following psuedocode will determine if the `available` version can satisfy the `requested` version:
```

Comment:
There's something a little funny about this, because it's not necessarily the case that a consumer implementation literally has "one spec version". But I guess on an abstract level you can consider a consumer that understands v1.2, v2.3, and v3.4 to actually be three separate implementations that happen to be identical.

Reply:
well, they might not even be identical! In the consumer, I'm imagining we'll have a mechanism like:

```rust
fn register(registry: &mut Registry) {
  registry.provide(spec: "https://specs.apollo.dev/cs/0.8", &apollo::composed::v0_8);
  registry.provide(spec: "https://specs.apollo.dev/cs/0.9", &apollo::composed::v0_9);
  registry.provide(spec: "https://specs.apollo.dev/cs/1.8", &apollo::composed::v1_8);
  registry.provide(spec: "https://specs.apollo.dev/cs/2.4", &apollo::composed::v2);
}
```

Then we have a general function that scans `@using` directives, figures out that to satisfy e.g. `cs/1.2` it should activate `apollo::composed::v1_8`, and returns a list of implementations to activate and what prefixes to activate them with.